server/backend/api/app/crud/crud_access_control_privilege.py

```diff
@@ -45,3 +45,23 @@ def get_delegated_admin_by_user_id(
         )
         .all()
     )
+
+
+def get_delegated_admin_by_user_id_and_role_id(
```

Comment:
I'm not keen on how this is returning a random item (if there are multiple matches). Would this make more sense as is_delegated_admin_... returning a boolean?

Reply:
Also, why are we checking this for a specific role? I'm missing the use case for this check right now.

Reply:
Yes, Basil's suggestion is really good.
Since all these are used for `router_guards` to determine delegated admin or not, and not directly needed for the object return, with boolean type of "**is_delegated_admin**()" make a lot sense, but not sure if Python support method overloadding... (probably not).
In the sqlalchemy query, something like `....filter().scalar() is not None` might do the work.